tests/rt.sh

```diff
@@ -97,12 +97,14 @@ rt_trap() {
 }
 
 cleanup() {
-  rm -rf ${LOCKDIR}
+  [[ ${delete_lock:-FALSE} == TRUE ]] && rm -rf ${LOCKDIR}
```

Comment:
Instead of relying on this variable (delete_lock) to be properly initialized and to have desired value I think it's better to look at the pid number of a process that created the lock file, and remove the lock only if it is the same as the pid of the current process.

For example:
```
    [[ $(awk '{print $2}' < "${LOCKDIR}/PID") == $$ ]] && rm -rf ${LOCKDIR}
```
Which better captures the intention that the lock file can be removed **only** be the process that created it.

Reply:
This new safeguard works, so I committed it.